transport/helpers/network/libp2p_impl.go

```diff
@@ -203,9 +236,47 @@ func (dtnet *libp2pDataTransferNetwork) SendMessage(
 	return s.Close()
 }
 
-func (dtnet *libp2pDataTransferNetwork) SetDelegate(r Receiver) {
-	dtnet.receiver = r
-	for _, p := range dtnet.dtProtocols {
+func (dtnet *libp2pDataTransferNetwork) SetDelegate(transportID datatransfer.TransportID, versions []datatransfer.Version, r Receiver) {
+	transportProtocols := transportProtocols{}
+	for _, dtProtocol := range dtnet.dtProtocols {
+		messageVersion, _ := MessageVersion(dtProtocol)
+		if isLegacyProtocol(dtProtocol) {
+
+			if transportID == datatransfer.LegacyTransportID {
```

Comment:
There's no check here for if `LegacyTransportVersion` is in the list of `versions`, so does that mean that any registration of `"graphsync"` will set up a legacy protocol regardless of what versions are provided? so there's no way to use this with `"graphsync"` without also wiring up for legacy? Probably not an immediate concern but we'll presumably want to phase that out at some point in the future.